app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class Api::V1::UsersController < ApplicationController
+  before_action :authorized, only: [:auto_login]
+
+  # REGISTER
+  def create
+    @user = User.create(user_params)
+    if @user.valid?
+      token = encode_token({ user_id: @user.id })
+      render json: { user: @user, token: token }
+    else
+      render json: { error: 'Invalid username or password' }, status: :unprocessable_entity
+    end
+  end
+
+  # LOGGING IN
```

Comment:
REST is broken. For login better to use SessionsController (in devise gem already implemented)